src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -140,17 +162,21 @@ public static PhysicalAddress Parse(string address)
             {
                 int value = (int)address[i];
 
-                if (value >= 0x30 && value <= 0x39)
+                if (value >= (int)'0' && value <= (int)'9')
                 {
                     value -= 0x30;
                 }
-                else if (value >= 0x41 && value <= 0x46)
+                else if (value >= (int)'A' && value <= (int)'F')
                 {
                     value -= 0x37;
```

Comment:
`value -= (int)'A';`

Reply:
@stephentoub This actually isn't quite right. `(int)'A'` resolves to 0x41 instead of 0x37. Basically, I'd need to also add an integer 10 to the value offset for a-f and A-F. Right now, subtracting `(int)'A'` directly would give value 0-5 for A-F, instead of 10-15.

Reply:
Yes, that was a typo.  Fixed.